PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/BeforeRunTaskUtils.java

```diff
@@ -0,0 +1,200 @@
+                                                                    @NotNull ExternalProjectPojo gradleProject,
+                                                                    RunConfiguration runConfiguration, boolean add)
+            throws IllegalAccessException {
+        updateBeforeRunOption(runConfigurationEditorComponent, ExternalSystemBeforeRunTask.class, task ->
+                                      Objects.nonNull(task) &&
+                                              Objects.nonNull(task.getTaskExecutionSettings())
+                                              && StringUtils.equals(task.getTaskExecutionSettings()
+                                                                        .getExternalProjectPath(),
+                                                                    gradleProject.getPath())
+                                              && CollectionUtils.isEqualCollection(task.getTaskExecutionSettings().getTaskNames(),
+                                                                                   Arrays.asList(GRADLE_TASK_ASSEMBLE)),
+            () -> {
+                GradleBeforeRunTaskProvider provider = new GradleBeforeRunTaskProvider(
+                        runConfiguration.getProject());
+                ExternalSystemBeforeRunTask task = provider.createTask(runConfiguration);
+                task.getTaskExecutionSettings()
+                    .setExternalSystemIdString(GradleConstants.SYSTEM_ID.toString());
+                task.getTaskExecutionSettings().setExternalProjectPath(gradleProject.getPath());
+                task.getTaskExecutionSettings().setTaskNames(Arrays.asList(GRADLE_TASK_ASSEMBLE));
+                return task;
+            }, runConfiguration, add);
+    }
+
+    public static <T extends BeforeRunTask> void updateBeforeRunOption(
+            @NotNull JComponent runConfigurationEditorComponent,
+            @NotNull Class<T> runTaskClass,
+            @NotNull Predicate<T> filter,
+            Producer<T> producer,
+            RunConfiguration runConfiguration,
+            boolean add) throws IllegalAccessException {
+        DataContext dataContext = DataManager.getInstance().getDataContext(runConfigurationEditorComponent);
+        ConfigurationSettingsEditorWrapper editor = ConfigurationSettingsEditorWrapper.CONFIGURATION_EDITOR_KEY.getData(
+                dataContext);
+        if (editor != null) {
+            List<T> tasks = ContainerUtil.findAll(editor.getStepsBeforeLaunch(), runTaskClass)
+                                         .stream()
+                                         .filter(filter)
+                                         .collect(Collectors.toList());
+            if (add && tasks.isEmpty()) {
+                T task = producer.produce();
+                task.setEnabled(true);
+                tasks.add(task);
+
+                RunManagerEx manager = RunManagerEx.getInstanceEx(runConfiguration.getProject());
```

Comment:
Please comment why we need to add the `task2`

Reply:
Done